### PR TITLE
Protocol Update: endpoints, TGT, error code, Svc Mgmt warning

### PR DESCRIPTION
**Pull request by mmoayyed**
This pull does the following:

1. Recommends the use of service management tool as a best practice to filter all service urls. 
2. Adds new CAS P3 endpoints
3. Fixes error codes and code formatting. 
4. Adds TGT to the CAS entities.

### CODE REVIEW

File: cas-server-documentation/protocol/CAS-Protocol-Specification.md [hunk 1]
```diff
@@ -1085,7 +1142,7 @@ and [2.5.1](<#head2.5.1>).
 -   The value of ticket-granting cookies SHALL contain adequate secure random data
     so that a ticket-granting cookie is not guessable in a reasonable period of time.
 
--   The value of ticket-granting cookies SHOULD begin with the characters "TGC-".
+-   The value of ticket-granting cookies SHOULD begin with the characters `TGC-`.
```
Comment (leleuj) [inline cas-server-documentation/protocol/CAS-Protocol-Specification.md]: We are currently using ```CASTGC```: should we move to TGC something?

**Comment by leleuj** [comment]
Great update. Thanks.

:+1:

File: cas-server-documentation/protocol/CAS-Protocol-Specification.md (hunk 1 above)
Comment (mmoayyed) [inline cas-server-documentation/protocol/CAS-Protocol-Specification.md]: Yes, excellent point. We should change the code to match the protocol.

Comment (battags) [inline cas-server-documentation/protocol/CAS-Protocol-Specification.md]: We're using CASTGC for the name of the cookie, not the prefix of the value, correct?  The prefix of the value is TGT.

Comment (mmoayyed) [inline cas-server-documentation/protocol/CAS-Protocol-Specification.md]: You're right. The name is specified as CASTGC. Maybe we could enforce both:

- *name* of the cookie should be CASTGC. 
- value of the cookie should begin with TGT, etc.

**Comment by mmoayyed** [comment]
Could I assume this pull is ready to go?

**Comment by robertoschwald** [comment]
Shouldn't we raise the version to 3.0.1 and change the release date? Otherwise people I think get confused.

**Comment by mmoayyed** [comment]
Sure. Good idea. I'll make the change to reflect that.

**Comment by mmoayyed** [comment]
Updated version and release date (v3.0.1)

**Comment by robertoschwald** [comment]
+1 Thanks for the work!

**Pull request merged**